x-pack/plugins/endpoint/public/embeddables/resolver/store/middleware.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Dispatch, MiddlewareAPI } from 'redux';
+import { KibanaReactContextValue } from '../../../../../../../src/plugins/kibana_react/public';
+import { EndpointPluginServices } from '../../../plugin';
+import { ResolverState, ResolverAction } from '../types';
+
+type MiddlewareFactory<S = ResolverState> = (
+  context?: KibanaReactContextValue<EndpointPluginServices>
+) => (
+  api: MiddlewareAPI<Dispatch<ResolverAction>, S>
+) => (next: Dispatch<ResolverAction>) => (action: ResolverAction) => unknown;
+
+export const resolverMiddlewareFactory: MiddlewareFactory = context => {
+  return api => next => async (action: ResolverAction) => {
+    next(action);
+    if (action.type === 'userChangedSelectedEvent') {
+      if (context?.services.http) {
+        api.dispatch({ type: 'appRequestedResolverData' });
+        const uniquePid = action.payload.selectedEvent?.endgame.unique_pid;
+        const legacyEndpointID = action.payload.selectedEvent?.agent.id;
```

Comment:
maybe validate that you have the params before triggering the `appRequested...` action.

Reply:
how would error handling work anyway from a user perspective?